### PR TITLE
feat: add client_options support for api endpoint override

### PR DESCRIPTION
**Pull request by busunkim96**
There is no manipulation of the endpoint url passed (e.g., `storage.googleapis.com` -> `https://foo.googleapis.com` but am happy to be convinced to do otherwise.

Replacement for #705.

Also sneaking in some kokoro config shuffling into this PR. (The status check named "Kokoro" is the real one.)

### CODE REVIEW

File: tests/test_discovery.py [hunk 1]
```diff
@@ -515,6 +515,25 @@ def test_building_with_developer_key_skips_adc(self):
...
 
+    def test_api_endpoint_override_from_client_options(self):
+        discovery = open(datafile("plus.json")).read()
+        api_endpoint = "https://foo.googleapis.com/"
+        options = google.api_core.client_options.ClientOptions(
+            api_endpoint=api_endpoint
+        )
+        plus = build_from_document(discovery, client_options=options)
+
+        self.assertEquals(plus._baseUrl, api_endpoint)
```
Comment (plamut) [inline tests/test_discovery.py]: (did not check the rest of the PR)

`assertEquals()` is deprecated, better to use `assertEqual()` instead.

File: googleapiclient/discovery.py [hunk 2]
```diff
@@ -350,7 +358,16 @@ def build_from_document(
         )
         raise InvalidJsonError()
 
-    base = urljoin(service["rootUrl"], service["servicePath"])
+    # If an API Endpoint is provided on client options, use that as the base URL
+    base = urljoin(service['rootUrl'], service["servicePath"])
+    if client_options:
+        if type(client_options) == dict:
```
[the remark this answers is not in the document]
Comment (busunkim96) [reply googleapiclient/discovery.py]: The API Core implementation currently specifies a dict (see https://googleapis.dev/python/google-api-core/latest/client_options.html). But this restriction is indeed arbitrary. Opened https://github.com/googleapis/python-api-core/issues/14 to track loosening this across the libraries.

File: tests/test_discovery.py (hunk 1 above)
Comment (busunkim96) [reply tests/test_discovery.py]: Done. Grepped through the files and found some other instances so I changed them en masse. Hopefully it doesn't make the diff to hard to read.

File: googleapiclient/discovery.py [hunk 3]
```diff
@@ -350,7 +358,16 @@ def build_from_document(
         )
         raise InvalidJsonError()
 
-    base = urljoin(service["rootUrl"], service["servicePath"])
+    # If an API Endpoint is provided on client options, use that as the base URL
+    base = urljoin(service['rootUrl'], service["servicePath"])
```
[the remark this answers is not in the document]
Comment (busunkim96) [reply googleapiclient/discovery.py]: I think it is OK for this to crash and burn if either of these files are missing. The `service` dict  isn't something we expect users to be constructing on their own. It's a JSON file generated automatically rom API definitions. See https://acceleratedmobilepageurl.googleapis.com/$discovery/rest?version=v1.

[the remark this answers is not in the document]
Comment (plamut) [reply googleapiclient/discovery.py]: I see, it's probably fine then.

File: googleapiclient/discovery.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (plamut) [reply googleapiclient/discovery.py]: I agree. If changing this, it should be done holistically across the entire library.

File: tests/test_discovery.py (hunk 1 above)
Comment (plamut) [reply tests/test_discovery.py]: The diff is fine if going through on a per-commit basis, thanks!

**Pull request merged**